src/app/malaria-101/activity-1/activity-1-2.component.ts

```diff
@@ -108,10 +112,12 @@ export class MalariaLifeCycleComponent implements OnInit {
             }
         }
 
-        if (!isWrongPos && arrLength === 6) {
+        if (!isWrongPos && arrLength === this.ARRAY_LENGTH) {
             this.activityComplete = true;
-          if (!this.completed) {const currStage = 3;
-            this._performanceService.openDialog(currStage); }
+          if (!this.completed) {
+            this._performanceService.openDialog(this.currStage);
+            this._leaderBoardService.updateLeaderBoard({activity: 'lifeCycle', level: 'level1'})
+          }
           this._sharedData.customSuccessAlert(this.alerts.activitySuccessMsg, this.alerts.activitySuccessTitle);
             this._dashboardService.updateProgressStatus(this._status).subscribe(response => {});
         } else if (arrLength === 6) {
```

Comment:
You need to add  6 and other constants as const.. It will improve code readability by lot.